### ISSUE TITLE
Extend regression in Datalab to handle predictions as input

### ISSUE THREAD

**Issue by elisno** [cleanlab/cleanlab#899]
This issue tracks work for handling `predictions` as inputs when finding label issues in Datlab for regression.

# Details
With #796 now merged, we can run label issue check for regression datasets.

The example shows how to take numerical features like so:

```python
from cleanlab import Datalab
import numpy as np

# Generate some toy-examples
N = 200
X = np.random.rand(N, 3)
y = 1 + 2 * X[:,0] + 3 * X[:,1] + 4 * X[:,2] + 0.05*np.random.randn(N)

# Run defaults
lab = Datalab(data={"y": y}, label_name='y', task="regression")
lab.find_issues(features=X)
```

We need to add a `predictions: Optional[np.ndarray] = None)` argument to https://github.com/cleanlab/cleanlab/blob/204651cf92301aa4d738cc2b5bdb9bd43125aa56/cleanlab/datalab/internal/issue_manager/regression/label.py#L87-L91

The idea is that something like this should work:

```python
from cleanlab import Datalab
import numpy as np

# Generate some toy-examples
N = 200
X = np.random.rand(N, 3)
y = 1 + 2 * X[:,0] + 3 * X[:,1] + 4 * X[:,2] + 0.05*np.random.randn(N)

# Run with existing predictions
from sklearn.neighbors import KNeighborsRegressor
# Not good modeling code, but just for example
y_pred = KNeighborsRegressor(n_neighbors=3).fit(X, y).predict(X)

lab = Datalab(data={"y": y}, label_name='y', task="regression")
lab.find_issues(predictions=y_pred, issue_types= {"label": {}})  # setting issue_types to focus on the issue manager.
```

When passing `predictions` to the LabelIssueManager for regression, the scores for examples should be computed with,

https://github.com/cleanlab/cleanlab/blob/0903c2a3614aa59dea6b826dfacbc995d5bf9c45/cleanlab/regression/rank.py#L38-L43

We still need to resolve how to flag examples with this approach (

## Note to contributors

When the argument is added to the issue manager, a quick sanity check would be to replace the last line in the code snippet above with:

```diff
-lab.find_issues(predictions=y_pred, issue_types= {"label": {}})
+lab.find_issues(issue_types={"label": {"predictions": y_pred}})
```

Defining the new `predictions` argument on the Datalab level will be the next logical step.

Both of these things must be done in the same PR.

🚨 **Before starting work on this issue** 🚨 , please take a moment to add your comments and thoughts regarding any unresolved specifications or questions.

**Comment by elisno** [comment]
#902 closed this issue.
